manticore/models/sql/compiler.py

```diff
@@ -90,3 +94,17 @@ def __maybe_set_limits(self, with_limits):
     def _compile_in(self, node: lookups.In):
         lookup = InFunction(node.lhs, node.rhs)
         return lookup.as_sql(self, self.connection)
+
+    def _compile_dict(self, param_dict: dict):
```

Comment:
manticore also [supports](https://manual.manticoresearch.com/Searching/Options) index weights dict, that does not correspond to model field list.

If you wish to support universal dict compiling option, you need to move field list check to `QuerySet.options()`.

Reply:
I did not move the check of the right side to a higher level, because found a way how to check the left side in the compiler.

Reply:
Django checks field names in `filter()` call, not at queryset evaluation. This helps to find the cause of an error, not consequences.

Reply:
fixed